pybboxes/annotations/base.py

```diff
@@ -0,0 +1,389 @@
+        labels_dir : str
+            immediate parent directory that houses all the image annoatations
+        images_dir : str
+            immediate parent directory that houses all the images (we need corresponding images to labels to extract image dimensions)
+        classes_file : str
+            path to classes.txt that lists all the class labels used in the annotation
+        """
+
+        if self._annotation_type != "yolo":
+            raise TypeError(
+                f"this instance of Annotations can only process {self._annotation_type} annotation file(s)"
+            )
+
+        if not os.path.exists(classes_file):
+            raise FileNotFoundError(f"{classes_file} doesn't exist")
+
+        if not os.path.exists(labels_dir):
+            raise NotADirectoryError(f"{labels_dir} is not a valid directory")
+
+        if not os.path.exists(images_dir):
+            raise NotADirectoryError(f"{images_dir} is not a valid directory")
+
+        with open(classes_file, "r") as f:
+            self._class_names = [line.strip() for line in f.readlines()]
+
+        for filename in os.listdir(labels_dir):
+            if filename.endswith(".txt"):
+                if "classes" in filename:
+                    continue  # if this is classes label, we have to skip as it donot contains bounding boxes data
+
+                image_name = filename.replace(".txt", ".jpg")  # we are assuming jpg extension
+                if not os.path.exists(os.path.join(images_dir, image_name)):
+                    image_name = filename.replace(".jpg", ".jpeg")  # see if image with jpeg extension exits
+                    if not os.path.exists(image_name):
+                        raise FileNotFoundError(f"{image_name} not found in images directory")
+
+                image_size = get_image_size(os.path.join(images_dir, image_name))  # we need for yolo format
+
+                with open(os.path.join(labels_dir, filename), "r") as f:
+                    for line in f:
+                        parts = line.strip().split()
+                        label_id = int(parts[0])  # extract the class/label id
+                        x_c, y_c, w, h = map(float, parts[1:5])
+
+                        bbox = BoundingBox.from_yolo(x_c, y_c, w, h, image_size=image_size)
+
+                        annotation = Annotation(
+                            box=bbox,
+                            label_id=label_id,
+                            label_name=self.id2label(label_id),
+                            annotation_type="yolo",
+                            image_width=image_size[0],
+                            image_height=image_size[1],
+                        )
+
+                        if image_name not in self._objects.keys():
+                            self._objects[image_name] = [annotation]
+                        else:
+                            self._objects[image_name].append(annotation)
+
+    def persist_as_yolo(self, export_dir: str):
```

Comment:
can we rename the methods as `write_as_...` or `save_as_...` I think it would be better if we stick to the global naming convention with the other python packages. I'd suggest either `read/write` or `load/save`, both are fine I think.

Reply:
hello, I have made the revisions that you requested for